plugins/app-backend/src/service/router.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { resolve as resolvePath, dirname } from 'path';
+import { notFoundHandler } from '@backstage/backend-common';
+import express from 'express';
+import Router from 'express-promise-router';
+import { Logger } from 'winston';
+import { injectEnvConfig } from '../lib/config';
+
+export interface RouterOptions {
+  logger: Logger;
+  appPackageName?: string;
+  staticFallbackHandler?: express.Handler;
+}
+
+export async function createRouter(
+  options: RouterOptions,
+): Promise<express.Router> {
+  const appDistDir = resolvePath(
+    dirname(
+      __non_webpack_require__.resolve(`${options.appPackageName}/package.json`),
```

Comment:
instead of using a weird global, could we just add a `requireImpl?: WhateverRequireTypeIs` to `RouterOptions`, and have it default at runtime to the actual `require`? or, can we inject the actual app module or something?

Reply:
I think we should add a package path resolve utility to backend-common, or look into if we can solve this by changing the behaviour of `__dirname` in the build. Either way I don't think we want the long term solution to be to pass the actual resolve function into the plugin.